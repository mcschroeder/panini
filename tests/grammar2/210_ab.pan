import axioms

f210 : {s:string|?} -> unit
f210 = \s:string.
  let p = match s "ab" in
  assert p

f211 : {s:string|?} -> unit
f211 = \s:string.
  let c1 = charAt s 0 in
  let p1 = match c1 "a" in
  let _ = assert p1 in
  let c2 = charAt s 1 in  
  let p2 = match c2 "b" in
  let _ = assert p2 in
  let n = length s in 
  let p3 = eq n 2 in
  assert p3
