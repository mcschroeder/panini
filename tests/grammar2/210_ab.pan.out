f210 : {s:𝕊 | s ∈ ab} → 𝟙
f211 : {s:𝕊 | s ∈ ab} → 𝟙